modules/k8s/namespace_test.go

```diff
@@ -34,3 +35,26 @@ func TestNamespaces(t *testing.T) {
 	namespace := GetNamespace(t, options, namespaceName)
 	require.Equal(t, namespace.Name, namespaceName)
 }
+
+func TestNamespaceWithMetadata(t *testing.T) {
+	t.Parallel()
+
+	uniqueId := random.UniqueId()
+	namespaceName := strings.ToLower(uniqueId)
+	options := NewKubectlOptions("", "", namespaceName)
+	namespaceLabels := map[string]string{"foo": "bar"}
+	namespaceObjectMetaWithLabels := metav1.ObjectMeta{
+		Name:   namespaceName,
+		Labels: namespaceLabels,
+	}
+	CreateNamespaceWithMetadata(t, options, namespaceObjectMetaWithLabels)
+	defer func() {
+		DeleteNamespace(t, options, namespaceName)
+		namespace := GetNamespace(t, options, namespaceName)
+		require.Equal(t, namespace.Status.Phase, corev1.NamespaceTerminating)
```

Comment:
One more question: does this depend on timing? E.g., If this code happens to run a few seconds later, would the namespace status be something different? Or would `GetNamespace` even return an error because the namespace has been deleted?

Reply:
At first glance, it seems like `GetNamespace` could throw an error if the timing was off. The only valid namespace phases are `Active` and `Terminating`, so I suppose another timing issue could be if the namespace was still in the Active phase. 

This same logic is used in `TestNamespaces`, so maybe it's not a concern if that test hasn't been flakey. 

I suppose a more reliable solution would be to add a `WaitUntilNamespaceDeleted` method or something similar

Reply:
OK, let's see if it works!

Reply:
Note that we're still waiting on https://github.com/gruntwork-io/terratest/pull/756 to be resolved here too!